lti_consumer/lti.py

```diff
@@ -0,0 +1,281 @@
+    The dict must have the "@type" key with value equal to "Result",
+    "resultScore" key with value equal to a number [0, 1],
+    The "@context" key must be present, but we don't do anything with it.  And the "comment" key may be
+    present, in which case it must be a string.
+
+    Arguments:
+        json_str (unicode):  The body of the LTI 2.0 results service request, which is a JSON string
+
+    Returns:
+        (float, str):  (score, [optional]comment) if verification checks out
+
+    Raises:
+        LtiError: if verification fails
+    """
+    try:
+        json_obj = json.loads(json_str)
+    except (ValueError, TypeError):
+        msg = "Supplied JSON string in request body could not be decoded: {}".format(json_str)
+        log.info("[LTI] %s", msg)
+        raise LtiError(msg)
+
+    # the standard supports a list of objects, who knows why. It must contain at least 1 element, and the
+    # first element must be a dict
+    if not isinstance(json_obj, dict):
+        if isinstance(json_obj, list) and len(json_obj) >= 1 and isinstance(json_obj[0], dict):
+            json_obj = json_obj[0]
+        else:
+            msg = ("Supplied JSON string is a list that does not contain an object as the first element. {}"
+                   .format(json_str))
+            log.info("[LTI] %s", msg)
+            raise LtiError(msg)
+
+    # '@type' must be "Result"
+    result_type = json_obj.get("@type")
+    if result_type != "Result":
+        msg = "JSON object does not contain correct @type attribute (should be 'Result', is z{})".format(result_type)
+        log.info("[LTI] %s", msg)
+        raise LtiError(msg)
+
+    # '@context' must be present as a key
+    REQUIRED_KEYS = ["@context"]  # pylint: disable=invalid-name
+    for key in REQUIRED_KEYS:
+        if key not in json_obj:
+            msg = "JSON object does not contain required key {}".format(key)
+            log.info("[LTI] %s", msg)
+            raise LtiError(msg)
+
+    # 'resultScore' is not present.  If this was a PUT this means it's actually a DELETE according
+    # to the LTI spec.  We will indicate this by returning None as score, "" as comment.
+    # The actual delete will be handled by the caller
+    if "resultScore" not in json_obj:
+        return None, json_obj.get('comment', "")
+
+    # if present, 'resultScore' must be a number between 0 and 1 inclusive
+    try:
+        score = float(json_obj.get('resultScore', "unconvertable"))  # Check if float is present and the right type
+        if not 0 <= score <= 1:
+            msg = 'score value outside the permitted range of 0-1.'
+            log.info("[LTI] %s", msg)
+            raise LtiError(msg)
+    except (TypeError, ValueError) as err:
+        msg = "Could not convert resultScore to float: {}".format(err.message)
+        log.info("[LTI] %s", msg)
+        raise LtiError(msg)
+
+    return score, json_obj.get('comment', "")
+
+
+class Lti1Consumer(object):
```

Comment:
Documentation below says: 
We have also begun to add support for LTI 1.2/2.0

Name of this class is Lti1Consumer. Is there a better name we could use for this so it's obvious that this is the 1.1 specification?